In [2]:
# ライブラリのインポート
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [4]:
# 元となるデータの取得
df_origin = pd.read_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-3/user-3-nonull.csv")
df_origin.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          7007053259          NaN   
1           1                18876          7007072135  Eye Tracker   
2           2                35303          7007088562  Eye Tracker   
3           3                51944          7007105203  Eye Tracker   
4           4                68932          7007122191  Eye Tracker   

  Project name Export date Participant name Recording name Recording date  \
0          JD4  2025/07/18           user-3     Recording4     2025/07/18   
1          JD4  2025/07/18           user-3     Recording4     2025/07/18   
2          JD4  2025/07/18           user-3     Recording4     2025/07/18   
3          JD4  2025/07/18           user-3     Recording4     2025/07/18   
4          JD4  2025/07/18           user-3     Recording4     2025/07/18   

  Recording date UTC  ... Original Media height Eye movement type  \
0         2025/07/18  ...                   NaN               NaN   
1         2025/07/18  ...                   NaN      EyesNotFound   
2         2025/07/18  ...                   NaN      EyesNotFound   
3         2025/07/18  ...                   NaN      EyesNotFound   
4         2025/07/18  ...                   NaN      EyesNotFound   

   Eye movement event duration Eye movement type index Fixation point X  \
0                          NaN                     NaN              NaN   
1                       2444.0                     1.0              NaN   
2                       2444.0                     1.0              NaN   
3                       2444.0                     1.0              NaN   
4                       2444.0                     1.0              NaN   

  Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  \
0              NaN                         NaN                         NaN   
1              NaN                         NaN                         NaN   
2              NaN                         NaN                         NaN   
3              NaN                         NaN                         NaN   
4              NaN                         NaN                         NaN   

   Mouse position X  Mouse position Y  
0               NaN               NaN  
1               NaN               NaN  
2               NaN               NaN  
3               NaN               NaN  
4               NaN               NaN  

[5 rows x 84 columns]

# カラムの整理

In [5]:
nunique_series = df_origin.nunique(dropna=False).to_string()
print(nunique_series)

Unnamed: 0                                    17439
Recording timestamp                           17426
Computer timestamp                            17426
Sensor                                            3
Project name                                      1
Export date                                       1
Participant name                                  1
Recording name                                    1
Recording date                                    1
Recording date UTC                                1
Recording start time                              1
Recording start time UTC                          1
Recording duration                                1
Timeline name                                     1
Recording Fixation filter name                    1
Recording software version                        1
Recording resolution height                       1
Recording resolution width                        1
Recording monitor latency                         1
Average cali

In [6]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series = df_origin.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

In [7]:
# カラム名とそのユニーク値を出力
for col in single_unique_cols:
    unique_val = df_origin[col].unique()
    print(f"{col}: {unique_val}")


Project name: ['JD4']
Export date: ['2025/07/18']
Participant name: ['user-3']
Recording name: ['Recording4']
Recording date: ['2025/07/18']
Recording date UTC: ['2025/07/18']
Recording start time: ['16:19:39.048']
Recording start time UTC: ['07:19:39.048']
Recording duration: [307137]
Timeline name: ['Timeline1']
Recording Fixation filter name: ['Tobii I-VT (Fixation)']
Recording software version: ['24.21.435']
Recording resolution height: [1080]
Recording resolution width: [1920]
Recording monitor latency: [10]
Average calibration accuracy (mm): [13]
Average calibration accuracy (degrees): [1.16]
Average calibration accuracy (pixels): [45]
Average validation accuracy (mm): [11.6]
Average validation precision SD (mm): [2.9]
Average validation precision RMS (mm): [3.6]
Average validation accuracy (degrees): [1.07]
Average validation precision SD (degrees): [0.27]
Average validation precision RMS (degrees): [0.33]
Average validation accuracy (pixels): [41]
Average validation precision S

## カラムの要素が1つしかない変数の説明
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-3`|
|Recording name| 録画名|`Recording4`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:19:39.048`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:19:39.048`|
|Recording duration| 録画時間|`307137`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`3`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.16`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`45`pixels|
|Average validation accuracy | 平均バリデーション精度|`11.6`mm|
|Average validation precision SD | 平均バリデーション予測SD|`2.9`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`3.6`mm|
|Average validation accuracy |平均バリデーション精度| `1.07`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.27`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.33`degrees|
|Average validation accuracy |平均バリデーション精度| `41`pixels|
|Average validation precision SD |平均バリデーション予測SD| `10`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `12`pixels|


In [8]:
# 削除した新しいデータフレーム
df_cleaned = df_origin.drop(columns=single_unique_cols)
df_cleaned.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          7007053259          NaN   
1           1                18876          7007072135  Eye Tracker   
2           2                35303          7007088562  Eye Tracker   
3           3                51944          7007105203  Eye Tracker   
4           4                68932          7007122191  Eye Tracker   

   Eyetracker timestamp           Event Event value  Gaze point X  \
0                   NaN  RecordingStart         NaN           NaN   
1          7.025374e+09             NaN         NaN           NaN   
2          7.025391e+09             NaN         NaN           NaN   
3          7.025407e+09             NaN         NaN           NaN   
4          7.025424e+09             NaN         NaN           NaN   

   Gaze point Y  Gaze point left X  ...  Original Media height  \
0           NaN                NaN  ...                    NaN   
1           NaN                NaN  ...                    NaN   
2           NaN                NaN  ...                    NaN   
3           NaN                NaN  ...                    NaN   
4           NaN                NaN  ...                    NaN   

   Eye movement type  Eye movement event duration  Eye movement type index  \
0                NaN                          NaN                      NaN   
1       EyesNotFound                       2444.0                      1.0   
2       EyesNotFound                       2444.0                      1.0   
3       EyesNotFound                       2444.0                      1.0   
4       EyesNotFound                       2444.0                      1.0   

   Fixation point X  Fixation point Y  Fixation point X (MCSnorm)  \
0               NaN               NaN                         NaN   
1               NaN               NaN                         NaN   
2               NaN               NaN                         NaN   
3               NaN               NaN                         NaN   
4               NaN               NaN                         NaN   

   Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
0                         NaN               NaN               NaN  
1                         NaN               NaN               NaN  
2                         NaN               NaN               NaN  
3                         NaN               NaN               NaN  
4                         NaN               NaN               NaN  

[5 rows x 57 columns]

## ユニークな値を取り除いたデータフレーム

In [9]:
print(df_cleaned.nunique(dropna=False).to_string())

Unnamed: 0                             17439
Recording timestamp                    17426
Computer timestamp                     17426
Sensor                                     3
Eyetracker timestamp                   17193
Event                                      9
Event value                                9
Gaze point X                            1749
Gaze point Y                            1174
Gaze point left X                       1741
Gaze point left Y                       1145
Gaze point right X                      1747
Gaze point right Y                      1170
Gaze direction left X                  12109
Gaze direction left Y                  11491
Gaze direction left Z                   5589
Gaze direction right X                 12257
Gaze direction right Y                 11416
Gaze direction right Z                  5270
Pupil diameter left                     1540
Pupil diameter right                    1565
Pupil diameter filtered                 1531
Validity l

## Sensor事にデータフレームを分割

In [10]:
# NaNを含む行
df_sensor_nan = df_cleaned[df_cleaned['Sensor'].isna()]

# "Mouse" の行
df_sensor_mouse = df_cleaned[df_cleaned['Sensor'] == "Mouse"]

# "Eye Tracker" の行
df_sensor_eye = df_cleaned[df_cleaned['Sensor'] == "Eye Tracker"]


### 問題の既知代わりの時間を取得する

In [11]:
df_sensor_nan

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
0               0                    0          7007053259    NaN   
176           176              2963138          7010016397    NaN   
6396         6396            106706705          7113759964    NaN   
6407         6407            106840587          7113893846    NaN   
7837         7837            147271055          7154324314    NaN   
7842         7842            147347035          7154400294    NaN   
7864         7864            147773983          7154827242    NaN   
7865         7865            147773983          7154827242    NaN   
8166         8166            152765643          7159818902    NaN   
8167         8167            152765643          7159818902    NaN   
10135       10135            185491835          7192545094    NaN   
10138       10138            185526232          7192579491    NaN   
10139       10139            185526232          7192579491    NaN   
10440       10440            190526180          7197579439    NaN   
10441       10441            190526180          7197579439    NaN   
12485       12485            224568384          7231621643    NaN   
12488       12488            224603689          7231656948    NaN   
12489       12489            224603689          7231656948    NaN   
12786       12786            229615756          7236669015    NaN   
12787       12787            229615756          7236669015    NaN   
14791       14791            262732636          7269785895    NaN   
14794       14794            262763816          7269817075    NaN   
14795       14795            262763816          7269817075    NaN   
15090       15090            267771675          7274824934    NaN   
15091       15091            267771675          7274824934    NaN   
17132       17132            301875926          7308929185    NaN   
17135       17135            301907892          7308961151    NaN   
17136       17136            301907892          7308961151    NaN   
17433       17433            306915856          7313969115    NaN   
17438       17438            307136991          7314190250    NaN   

       Eyetracker timestamp                          Event Event value  \
0                       NaN                 RecordingStart         NaN   
176                     NaN  Eye tracker Calibration start         NaN   
6396                    NaN                     MouseEvent  Down, Left   
6407                    NaN                     MouseEvent    Up, Left   
7837                    NaN                     MouseEvent  Down, Left   
7842                    NaN                     MouseEvent    Up, Left   
7864                    NaN    Eye tracker Calibration end         NaN   
7865                    NaN             ImageStimulusStart      blank2   
8166                    NaN               ImageStimulusEnd      blank2   
8167                    NaN             ImageStimulusStart  視線誘導問題.001   
10135                   NaN                  KeyboardEvent       Space   
10138                   NaN               ImageStimulusEnd  視線誘導問題.001   
10139                   NaN             ImageStimulusStart      blank2   
10440                   NaN               ImageStimulusEnd      blank2   
10441                   NaN             ImageStimulusStart  視線誘導問題.002   
12485                   NaN                  KeyboardEvent       Space   
12488                   NaN               ImageStimulusEnd  視線誘導問題.002   
12489                   NaN             ImageStimulusStart      blank2   
12786                   NaN               ImageStimulusEnd      blank2   
12787                   NaN             ImageStimulusStart  視線誘導問題.003   
14791                   NaN                  KeyboardEvent       Space   
14794                   NaN               ImageStimulusEnd  視線誘導問題.003   
14795                   NaN             ImageStimulusStart      blank2   
15090                   NaN               ImageStimulusEnd      blank2   
15091            

In [12]:
# "視線誘導問題." を含む行を抽出
df_filtered = df_sensor_nan[df_sensor_nan["Event value"].astype(str).str.contains("視線誘導問題.")].copy()

# 結果の確認（先頭5行を表示）
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']]
print(df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']])

       Recording timestamp  Computer timestamp Event value
8167             152765643          7159818902  視線誘導問題.001
10138            185526232          7192579491  視線誘導問題.001
10441            190526180          7197579439  視線誘導問題.002
12488            224603689          7231656948  視線誘導問題.002
12787            229615756          7236669015  視線誘導問題.003
14794            262763816          7269817075  視線誘導問題.003
15091            267771675          7274824934  視線誘導問題.004
17135            301907892          7308961151  視線誘導問題.004


### 各問題の開始時刻
|       |Recording timestamp|  Computer timestamp| Event value|
|-------|-------------------|--------------------|------------|
|8167|152765643|7159818902|視線誘導問題.001|
|10138|185526232|7192579491|視線誘導問題.001|
|10441|190526180|7197579439|視線誘導問題.002|
|12488|224603689|7231656948|視線誘導問題.002|
|12787|229615756|7236669015|視線誘導問題.003|
|14794|262763816|7269817075|視線誘導問題.003|
|15091|267771675|7274824934|視線誘導問題.004|
|17135|301907892|7308961151|視線誘導問題.004|


In [13]:
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-3/start-stop-timing.csv")

In [14]:
df_sensor_mouse

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
6239         6239            105174802          7112228061  Mouse   
6241         6241            105192227          7112245486  Mouse   
6243         6243            105208283          7112261542  Mouse   
6245         6245            105224248          7112277507  Mouse   
6247         6247            105240241          7112293500  Mouse   
...           ...                  ...                 ...    ...   
14778       14778            262600072          7269653331  Mouse   
14783       14783            262655908          7269709167  Mouse   
14785       14785            262677645          7269730904  Mouse   
14787       14787            262697512          7269750771  Mouse   
14790       14790            262718349          7269771608  Mouse   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
6239                    NaN   NaN         NaN           NaN           NaN   
6241                    NaN   NaN         NaN           NaN           NaN   
6243                    NaN   NaN         NaN           NaN           NaN   
6245                    NaN   NaN         NaN           NaN           NaN   
6247                    NaN   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
14778                   NaN   NaN         NaN           NaN           NaN   
14783                   NaN   NaN         NaN           NaN           NaN   
14785                   NaN   NaN         NaN           NaN           NaN   
14787                   NaN   NaN         NaN           NaN           NaN   
14790                   NaN   NaN         NaN           NaN           NaN   

       Gaze point left X  ...  Original Media height  Eye movement type  \
6239                 NaN  ...                    NaN           Fixation   
6241                 NaN  ...                    NaN           Fixation   
6243                 NaN  ...                    NaN            Saccade   
6245                 NaN  ...                    NaN       EyesNotFound   
6247                 NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
14778                NaN  ...                 1080.0       EyesNotFound   
14783                NaN  ...                 1080.0       EyesNotFound   
14785                NaN  ...                 1080.0       EyesNotFound   
14787                NaN  ...                 1080.0       EyesNotFound   
14790                NaN  ...                 1080.0       Unclassified   

       Eye movement event duration  Eye movement type index  Fixation point X  \
6239                         100.0                    195.0            1012.0   
6241                         100.0                    195.0            1012.0   
6243                          17.0                    297.0               NaN   
6245                         116.0                    100.0               NaN   
6247                         116.0                    100.0               NaN   
...                            ...                      ...               ...   
14778                        166.0                    185.0               NaN   
14783                        166.0                    185.0               NaN   
14785                        166.0                    185.0               NaN   
14787                        166.0                    185.0               NaN   
14790                         33.0                    165.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
6239              540.0                         NaN   
6241              540.0                         NaN   
6243                NaN                         NaN   
6245                NaN                         NaN   
6247                NaN                         NaN   
...                 ...                         .

In [15]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
1               1                18876          7007072135  Eye Tracker   
2               2                35303          7007088562  Eye Tracker   
3               3                51944          7007105203  Eye Tracker   
4               4                68932          7007122191  Eye Tracker   
5               5                85272          7007138531  Eye Tracker   
...           ...                  ...                 ...          ...   
17432       17432            306904748          7313958007  Eye Tracker   
17434       17434            306921389          7313974648  Eye Tracker   
17435       17435            306938388          7313991647  Eye Tracker   
17436       17436            306954671          7314007930  Eye Tracker   
17437       17437            306971312          7314024571  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
1              7.025374e+09   NaN         NaN           NaN           NaN   
2              7.025391e+09   NaN         NaN           NaN           NaN   
3              7.025407e+09   NaN         NaN           NaN           NaN   
4              7.025424e+09   NaN         NaN           NaN           NaN   
5              7.025441e+09   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
17432          7.332241e+09   NaN         NaN           NaN           NaN   
17434          7.332258e+09   NaN         NaN           NaN           NaN   
17435          7.332275e+09   NaN         NaN           NaN           NaN   
17436          7.332291e+09   NaN         NaN           NaN           NaN   
17437          7.332308e+09   NaN         NaN           NaN           NaN   

       Gaze point left X  ...  Original Media height  Eye movement type  \
1                    NaN  ...                    NaN       EyesNotFound   
2                    NaN  ...                    NaN       EyesNotFound   
3                    NaN  ...                    NaN       EyesNotFound   
4                    NaN  ...                    NaN       EyesNotFound   
5                    NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
17432                NaN  ...                 1080.0       EyesNotFound   
17434                NaN  ...                    NaN       EyesNotFound   
17435                NaN  ...                    NaN       EyesNotFound   
17436                NaN  ...                    NaN       EyesNotFound   
17437                NaN  ...                    NaN       EyesNotFound   

       Eye movement event duration  Eye movement type index  Fixation point X  \
1                           2444.0                      1.0               NaN   
2                           2444.0                      1.0               NaN   
3                           2444.0                      1.0               NaN   
4                           2444.0                      1.0               NaN   
5                           2444.0                      1.0               NaN   
...                            ...                      ...               ...   
17432                       2852.0                    221.0               NaN   
17434                       2852.0                    221.0               NaN   
17435                       2852.0                    221.0               NaN   
17436                       2852.0                    221.0               NaN   
17437                       2852.0                    221.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
1                   NaN                         NaN   
2                   NaN                         NaN   
3                   NaN                         NaN   
4                   NaN                         NaN   
5                   NaN         

## Mouse以外のデータを抽出

In [16]:
df_cleaned_ver2 = df_cleaned[df_cleaned['Sensor'] != "Mouse"]
nunique_series_ver2 = df_cleaned_ver2.nunique(dropna=False)
single_unique_cols_ver2 = nunique_series_ver2[nunique_series_ver2 == 1].index.tolist()
# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver2:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")

Mouse position X: [nan]
Mouse position Y: [nan]


In [17]:
df_cleaned_ver2 = df_cleaned_ver2.drop(columns=single_unique_cols_ver2)

In [18]:
print(df_cleaned_ver2.nunique(dropna=False).to_string())

Unnamed: 0                             17222
Recording timestamp                    17213
Computer timestamp                     17213
Sensor                                     2
Eyetracker timestamp                   17193
Event                                      9
Event value                                9
Gaze point X                            1749
Gaze point Y                            1174
Gaze point left X                       1741
Gaze point left Y                       1145
Gaze point right X                      1747
Gaze point right Y                      1170
Gaze direction left X                  12109
Gaze direction left Y                  11491
Gaze direction left Z                   5589
Gaze direction right X                 12257
Gaze direction right Y                 11416
Gaze direction right Z                  5270
Pupil diameter left                     1540
Pupil diameter right                    1565
Pupil diameter filtered                 1531
Validity l

In [19]:
df_cleaned_ver2.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          7007053259          NaN   
1           1                18876          7007072135  Eye Tracker   
2           2                35303          7007088562  Eye Tracker   
3           3                51944          7007105203  Eye Tracker   
4           4                68932          7007122191  Eye Tracker   

   Eyetracker timestamp           Event Event value  Gaze point X  \
0                   NaN  RecordingStart         NaN           NaN   
1          7.025374e+09             NaN         NaN           NaN   
2          7.025391e+09             NaN         NaN           NaN   
3          7.025407e+09             NaN         NaN           NaN   
4          7.025424e+09             NaN         NaN           NaN   

   Gaze point Y  Gaze point left X  ...  Presented Media position Y (DACSpx)  \
0           NaN                NaN  ...                                  NaN   
1           NaN                NaN  ...                                  NaN   
2           NaN                NaN  ...                                  NaN   
3           NaN                NaN  ...                                  NaN   
4           NaN                NaN  ...                                  NaN   

   Original Media width  Original Media height  Eye movement type  \
0                   NaN                    NaN                NaN   
1                   NaN                    NaN       EyesNotFound   
2                   NaN                    NaN       EyesNotFound   
3                   NaN                    NaN       EyesNotFound   
4                   NaN                    NaN       EyesNotFound   

   Eye movement event duration  Eye movement type index  Fixation point X  \
0                          NaN                      NaN               NaN   
1                       2444.0                      1.0               NaN   
2                       2444.0                      1.0               NaN   
3                       2444.0                      1.0               NaN   
4                       2444.0                      1.0               NaN   

   Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  
0               NaN                         NaN                         NaN  
1               NaN                         NaN                         NaN  
2               NaN                         NaN                         NaN  
3               NaN                         NaN                         NaN  
4               NaN                         NaN                         NaN  

[5 rows x 55 columns]

In [20]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series_ver3 = df_cleaned_ver2.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols_ver3 = nunique_series_ver3[nunique_series_ver3 == 2].index.tolist()

# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver3:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")


Sensor: [nan 'Eye Tracker']
Presented Media width: [  nan 1920.]
Presented Media height: [  nan 1080.]
Presented Media position X (DACSpx): [nan  0.]
Presented Media position Y (DACSpx): [nan  0.]
Original Media width: [  nan 1920.]
Original Media height: [  nan 1080.]


### Sensor=Mouse以外に着目した後に要素が少ない特徴量について
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Mouse position X| マウス位置 X|`nan`|
|Mouse position Y| マウス位置 Y|`nan`|
|Sensor| センサー|`nan`, `Eye Tracker`|
|Presented Media width|提示されたメディアの幅|`nan` `1920`|
|Presented Media height|提示されたメディアの高さ|`nan`, `1080`|
|Presented Media position X (DACSpx)| 提示されたメディア位置 X|`nan`,  `0`|
|Presented Media position Y (DACSpx)| 提示されたメディアの位置 Y|`nan`,  `0`|
|Original Media width| オリジナルメディアの幅|`nan`, `1920`|
|Original Media height| オリジナルメディアの高さ|`nan`,`1080`|

In [21]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
1               1                18876          7007072135  Eye Tracker   
2               2                35303          7007088562  Eye Tracker   
3               3                51944          7007105203  Eye Tracker   
4               4                68932          7007122191  Eye Tracker   
5               5                85272          7007138531  Eye Tracker   
...           ...                  ...                 ...          ...   
17432       17432            306904748          7313958007  Eye Tracker   
17434       17434            306921389          7313974648  Eye Tracker   
17435       17435            306938388          7313991647  Eye Tracker   
17436       17436            306954671          7314007930  Eye Tracker   
17437       17437            306971312          7314024571  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
1              7.025374e+09   NaN         NaN           NaN           NaN   
2              7.025391e+09   NaN         NaN           NaN           NaN   
3              7.025407e+09   NaN         NaN           NaN           NaN   
4              7.025424e+09   NaN         NaN           NaN           NaN   
5              7.025441e+09   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
17432          7.332241e+09   NaN         NaN           NaN           NaN   
17434          7.332258e+09   NaN         NaN           NaN           NaN   
17435          7.332275e+09   NaN         NaN           NaN           NaN   
17436          7.332291e+09   NaN         NaN           NaN           NaN   
17437          7.332308e+09   NaN         NaN           NaN           NaN   

       Gaze point left X  ...  Original Media height  Eye movement type  \
1                    NaN  ...                    NaN       EyesNotFound   
2                    NaN  ...                    NaN       EyesNotFound   
3                    NaN  ...                    NaN       EyesNotFound   
4                    NaN  ...                    NaN       EyesNotFound   
5                    NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
17432                NaN  ...                 1080.0       EyesNotFound   
17434                NaN  ...                    NaN       EyesNotFound   
17435                NaN  ...                    NaN       EyesNotFound   
17436                NaN  ...                    NaN       EyesNotFound   
17437                NaN  ...                    NaN       EyesNotFound   

       Eye movement event duration  Eye movement type index  Fixation point X  \
1                           2444.0                      1.0               NaN   
2                           2444.0                      1.0               NaN   
3                           2444.0                      1.0               NaN   
4                           2444.0                      1.0               NaN   
5                           2444.0                      1.0               NaN   
...                            ...                      ...               ...   
17432                       2852.0                    221.0               NaN   
17434                       2852.0                    221.0               NaN   
17435                       2852.0                    221.0               NaN   
17436                       2852.0                    221.0               NaN   
17437                       2852.0                    221.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
1                   NaN                         NaN   
2                   NaN                         NaN   
3                   NaN                         NaN   
4                   NaN                         NaN   
5                   NaN         

## 各問題毎のデータへ切り分け

In [22]:
# 結果を辞書に格納
extracted_data = {}

targets = df_filtered["Event value"].unique()
for target in targets:
    timestamps = df_filtered[df_filtered["Event value"] == target]["Recording timestamp"].iloc[0:2]
    if len(timestamps) < 2:
        continue  # start/endが不足している場合スキップ

    start, end = timestamps.iloc[0], timestamps.iloc[1]

    extracted = df_sensor_eye[
        (df_sensor_eye["Recording timestamp"] >= start) &
        (df_sensor_eye["Recording timestamp"] <= end)
    ].copy()

    extracted_data[target] = extracted  # 日本語のキーOK

In [24]:
extracted_data["視線誘導問題.001"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
8168        8168            152768808          7159822067  Eye Tracker   
8169        8169            152784905          7159838164  Eye Tracker   
8170        8170            152801763          7159855022  Eye Tracker   
8171        8171            152818731          7159871990  Eye Tracker   
8172        8172            152834874          7159888133  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
8168          7.178115e+09   NaN         NaN         947.0         449.0   
8169          7.178131e+09   NaN         NaN         950.0         453.0   
8170          7.178148e+09   NaN         NaN         952.0         450.0   
8171          7.178165e+09   NaN         NaN         953.0         452.0   
8172          7.178181e+09   NaN         NaN         956.0         446.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
8168              936.0  ...                 1080.0           Fixation   
8169              941.0  ...                 1080.0           Fixation   
8170              948.0  ...                 1080.0           Fixation   
8171              954.0  ...                 1080.0           Fixation   
8172              953.0  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
8168                        815.0                    259.0             950.0   
8169                        815.0                    259.0             950.0   
8170                        815.0                    259.0             950.0   
8171                        815.0                    259.0             950.0   
8172                        815.0                    259.0             950.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
8168             452.0                      0.4949   
8169             452.0                      0.4949   
8170             452.0                      0.4949   
8171             452.0                      0.4949   
8172             452.0                      0.4949   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
8168                      0.4184               NaN               NaN  
8169                      0.4184               NaN               NaN  
8170                      0.4184               NaN               NaN  
8171                      0.4184               NaN               NaN  
8172                      0.4184               NaN               NaN  

[5 rows x 57 columns]

In [25]:
print(extracted_data["視線誘導問題.001"].nunique(dropna=False).to_string())

Unnamed: 0                             1969
Recording timestamp                    1969
Computer timestamp                     1969
Sensor                                    1
Eyetracker timestamp                   1969
Event                                     1
Event value                               1
Gaze point X                            808
Gaze point Y                            555
Gaze point left X                       834
Gaze point left Y                       554
Gaze point right X                      817
Gaze point right Y                      542
Gaze direction left X                  1846
Gaze direction left Y                  1806
Gaze direction left Z                  1613
Gaze direction right X                 1857
Gaze direction right Y                 1834
Gaze direction right Z                 1570
Pupil diameter left                     532
Pupil diameter right                    555
Pupil diameter filtered                 534
Validity left                   

In [ ]:
extracted_data["視線誘導問題.001"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
7058        7058            131505488          6790459509  Eye Tracker   
7059        7059            131521504          6790475525  Eye Tracker   
7060        7060            131538250          6790492271  Eye Tracker   
7061        7061            131555411          6790509432  Eye Tracker   
7062        7062            131571376          6790525397  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
7058          6.808775e+09   NaN         NaN        1168.0         517.0   
7059          6.808791e+09   NaN         NaN        1168.0         515.0   
7060          6.808808e+09   NaN         NaN        1173.0         510.0   
7061          6.808825e+09   NaN         NaN        1172.0         509.0   
7062          6.808841e+09   NaN         NaN        1084.0         487.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
7058             1168.0  ...                 1080.0           Fixation   
7059             1167.0  ...                 1080.0           Fixation   
7060             1168.0  ...                 1080.0           Fixation   
7061             1163.0  ...                 1080.0           Fixation   
7062             1075.0  ...                 1080.0            Saccade   

      Eye movement event duration  Eye movement type index  Fixation point X  \
7058                        216.0                    163.0            1175.0   
7059                        216.0                    163.0            1175.0   
7060                        216.0                    163.0            1175.0   
7061                        216.0                    163.0            1175.0   
7062                         50.0                    215.0               NaN   

      Fixation point Y  Fixation point X (MCSnorm)  \
7058             511.0                      0.6122   
7059             511.0                      0.6122   
7060             511.0                      0.6122   
7061             511.0                      0.6122   
7062               NaN                         NaN   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
7058                      0.4728               NaN               NaN  
7059                      0.4728               NaN               NaN  
7060                      0.4728               NaN               NaN  
7061                      0.4728               NaN               NaN  
7062                         NaN               NaN               NaN  

[5 rows x 57 columns]

In [26]:
extracted_data["視線誘導問題.002"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
10442       10442            190527143          7197580402  Eye Tracker   
10443       10443            190543903          7197597162  Eye Tracker   
10444       10444            190560839          7197614098  Eye Tracker   
10445       10445            190577016          7197630275  Eye Tracker   
10446       10446            190593823          7197647082  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
10442          7.215871e+09   NaN         NaN         937.0         439.0   
10443          7.215888e+09   NaN         NaN         934.0         438.0   
10444          7.215905e+09   NaN         NaN         929.0         442.0   
10445          7.215921e+09   NaN         NaN         934.0         444.0   
10446          7.215938e+09   NaN         NaN         932.0         440.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
10442              924.0  ...                 1080.0           Fixation   
10443              924.0  ...                 1080.0           Fixation   
10444              927.0  ...                 1080.0           Fixation   
10445              927.0  ...                 1080.0           Fixation   
10446              926.0  ...                 1080.0           Fixation   

       Eye movement event duration  Eye movement type index  Fixation point X  \
10442                       2413.0                    375.0             934.0   
10443                       2413.0                    375.0             934.0   
10444                       2413.0                    375.0             934.0   
10445                       2413.0                    375.0             934.0   
10446                       2413.0                    375.0             934.0   

       Fixation point Y  Fixation point X (MCSnorm)  \
10442             462.0                      0.4867   
10443             462.0                      0.4867   
10444             462.0                      0.4867   
10445             462.0                      0.4867   
10446             462.0                      0.4867   

       Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
10442                      0.4279               NaN               NaN  
10443                      0.4279               NaN               NaN  
10444                      0.4279               NaN               NaN  
10445                      0.4279               NaN               NaN  
10446                      0.4279               NaN               NaN  

[5 rows x 57 columns]

In [27]:
extracted_data["視線誘導問題.003"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
12788       12788            229631258          7236684517  Eye Tracker   
12789       12789            229647867          7236701126  Eye Tracker   
12790       12790            229664793          7236718052  Eye Tracker   
12791       12791            229681122          7236734381  Eye Tracker   
12792       12792            229697776          7236751035  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
12788          7.254973e+09   NaN         NaN         872.0         386.0   
12789          7.254989e+09   NaN         NaN         872.0         388.0   
12790          7.255006e+09   NaN         NaN         865.0         395.0   
12791          7.255022e+09   NaN         NaN         866.0         390.0   
12792          7.255039e+09   NaN         NaN         867.0         387.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
12788              858.0  ...                 1080.0           Fixation   
12789              852.0  ...                 1080.0           Fixation   
12790              848.0  ...                 1080.0           Fixation   
12791              846.0  ...                 1080.0           Fixation   
12792              842.0  ...                 1080.0           Fixation   

       Eye movement event duration  Eye movement type index  Fixation point X  \
12788                        499.0                    487.0             866.0   
12789                        499.0                    487.0             866.0   
12790                        499.0                    487.0             866.0   
12791                        499.0                    487.0             866.0   
12792                        499.0                    487.0             866.0   

       Fixation point Y  Fixation point X (MCSnorm)  \
12788             400.0                      0.4508   
12789             400.0                      0.4508   
12790             400.0                      0.4508   
12791             400.0                      0.4508   
12792             400.0                      0.4508   

       Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
12788                      0.3704               NaN               NaN  
12789                      0.3704               NaN               NaN  
12790                      0.3704               NaN               NaN  
12791                      0.3704               NaN               NaN  
12792                      0.3704               NaN               NaN  

[5 rows x 57 columns]

In [28]:
for key in extracted_data.keys():
    print(f"Key: {key}")
    df = extracted_data[key]
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()
    for col in single_unique_cols:
        unique_val = df[col].unique()
        print(f"  {col}: {unique_val}")

Key: 視線誘導問題.001
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.001']
  Presented Media name: ['視線誘導問題.001.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.002
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.002']
  Presented Media name: ['視線誘導問題.002.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.003
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.003']
  Presen

In [29]:
# 空の辞書に結果を格納
cleaned_data = {}

for key, df in extracted_data.items():
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

    # そのカラムを除外して新しいデータフレームを作成
    df_cleaned = df.drop(columns=single_unique_cols).copy()

    cleaned_data[key] = df_cleaned

    print(f"{key}: Removed {len(single_unique_cols)} columns -> Remaining {df_cleaned.shape[1]} columns")


視線誘導問題.001: Removed 13 columns -> Remaining 44 columns
視線誘導問題.002: Removed 13 columns -> Remaining 44 columns
視線誘導問題.003: Removed 13 columns -> Remaining 44 columns
視線誘導問題.004: Removed 13 columns -> Remaining 44 columns


### カラムの要素が1つしかない変数の説明 写真1
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-3`|
|Recording name| 録画名|`Recording4`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:19:39.048`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:19:39.048`|
|Recording duration| 録画時間|`307137`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`3`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.16`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`45`pixels|
|Average validation accuracy | 平均バリデーション精度|`11.6`mm|
|Average validation precision SD | 平均バリデーション予測SD|`2.9`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`3.6`mm|
|Average validation accuracy |平均バリデーション精度| `1.07`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.27`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.33`degrees|
|Average validation accuracy |平均バリデーション精度| `41`pixels|
|Average validation precision SD |平均バリデーション予測SD| `10`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `12`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.001`|
|Presented Media name|提示されたメディア名| `視線誘導問題.001.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [30]:
cleaned_data["視線誘導問題.001"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
8168         8168            152768808          7159822067   
8169         8169            152784905          7159838164   
8170         8170            152801763          7159855022   
8171         8171            152818731          7159871990   
8172         8172            152834874          7159888133   
...           ...                  ...                 ...   
10132       10132            185452503          7192505762   
10133       10133            185468749          7192522008   
10134       10134            185485284          7192538543   
10136       10136            185502326          7192555585   
10137       10137            185518672          7192571931   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
8168           7.178115e+09         947.0         449.0              936.0   
8169           7.178131e+09         950.0         453.0              941.0   
8170           7.178148e+09         952.0         450.0              948.0   
8171           7.178165e+09         953.0         452.0              954.0   
8172           7.178181e+09         956.0         446.0              953.0   
...                     ...           ...           ...                ...   
10132          7.210797e+09        1213.0         541.0             1209.0   
10133          7.210813e+09        1223.0         543.0             1202.0   
10134          7.210829e+09        1220.0         546.0             1200.0   
10136          7.210846e+09        1216.0         551.0             1189.0   
10137          7.210863e+09        1215.0         545.0             1199.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
8168               454.0               958.0               445.0  ...   
8169               452.0               958.0               454.0  ...   
8170               446.0               956.0               454.0  ...   
8171               453.0               952.0               451.0  ...   
8172               451.0               960.0               441.0  ...   
...                  ...                 ...                 ...  ...   
10132              527.0              1218.0               554.0  ...   
10133              531.0              1244.0               554.0  ...   
10134              538.0              1240.0               554.0  ...   
10136              546.0              1244.0               555.0  ...   
10137              539.0              1231.0               551.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
8168                        0.4200                        0.4991   
8169                        0.4190                        0.4989   
8170                        0.4133                        0.4981   
8171                        0.4192                        0.4960   
8172                        0.4180                        0.5000   
...                            ...                           ...   
10132                       0.4883                        0.6344   
10133                       0.4921                        0.6479   
10134                       0.4981                        0.6457   
10136                       0.5060                        0.6479   
10137                       0.4991                        0.6412   

       Gaze point right Y (MCSnorm)  Eye movement type  \
8168                         0.4121           Fixation   
8169                         0.4203           Fixation   
8170                         0.4208           Fixation   
8171                         0.4179           Fixation   
8172                         0.4084           Fixation   
...                             ...                ...   
10132                        0.5131           Fixation   
10133                        0.5130           Fixation   
10134                        0.5134           Fixation   
10136                        0.5140           Fixat

In [31]:
cleaned_data["視線誘導問題.001"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-3/user-3-pic-1.csv")

### カラムの要素が1つしかない変数の説明 写真2
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-3`|
|Recording name| 録画名|`Recording4`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:19:39.048`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:19:39.048`|
|Recording duration| 録画時間|`307137`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`3`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.16`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`45`pixels|
|Average validation accuracy | 平均バリデーション精度|`11.6`mm|
|Average validation precision SD | 平均バリデーション予測SD|`2.9`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`3.6`mm|
|Average validation accuracy |平均バリデーション精度| `1.07`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.27`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.33`degrees|
|Average validation accuracy |平均バリデーション精度| `41`pixels|
|Average validation precision SD |平均バリデーション予測SD| `10`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `12`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.002`|
|Presented Media name|提示されたメディア名| `視線誘導問題.002.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [32]:
cleaned_data["視線誘導問題.002"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
10442       10442            190527143          7197580402   
10443       10443            190543903          7197597162   
10444       10444            190560839          7197614098   
10445       10445            190577016          7197630275   
10446       10446            190593823          7197647082   
...           ...                  ...                 ...   
12482       12482            224534488          7231587747   
12483       12483            224551547          7231604806   
12484       12484            224568368          7231621627   
12486       12486            224584411          7231637670   
12487       12487            224601516          7231654775   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
10442          7.215871e+09         937.0         439.0              924.0   
10443          7.215888e+09         934.0         438.0              924.0   
10444          7.215905e+09         929.0         442.0              927.0   
10445          7.215921e+09         934.0         444.0              927.0   
10446          7.215938e+09         932.0         440.0              926.0   
...                     ...           ...           ...                ...   
12482          7.249876e+09         866.0         425.0              838.0   
12483          7.249893e+09         866.0         432.0              845.0   
12484          7.249910e+09         861.0         430.0              840.0   
12486          7.249926e+09         860.0         432.0              843.0   
12487          7.249943e+09         855.0         429.0              837.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
10442              443.0               950.0               436.0  ...   
10443              446.0               944.0               430.0  ...   
10444              452.0               932.0               432.0  ...   
10445              457.0               940.0               431.0  ...   
10446              450.0               939.0               430.0  ...   
...                  ...                 ...                 ...  ...   
12482              432.0               894.0               417.0  ...   
12483              452.0               887.0               412.0  ...   
12484              449.0               883.0               410.0  ...   
12486              440.0               878.0               424.0  ...   
12487              435.0               874.0               423.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
10442                       0.4101                        0.4948   
10443                       0.4131                        0.4918   
10444                       0.4187                        0.4852   
10445                       0.4231                        0.4897   
10446                       0.4171                        0.4890   
...                            ...                           ...   
12482                       0.3999                        0.4656   
12483                       0.4182                        0.4621   
12484                       0.4162                        0.4598   
12486                       0.4073                        0.4573   
12487                       0.4030                        0.4551   

       Gaze point right Y (MCSnorm)  Eye movement type  \
10442                        0.4037           Fixation   
10443                        0.3983           Fixation   
10444                        0.4002           Fixation   
10445                        0.3991           Fixation   
10446                        0.3984           Fixation   
...                             ...                ...   
12482                        0.3863           Fixation   
12483                        0.3816           Fixation   
12484                        0.3792           Fixation   
12486                        0.3924           Fixat

In [33]:
cleaned_data["視線誘導問題.002"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-3/user-3-pic-2.csv")

### カラムの要素が1つしかない変数の説明 写真3
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-3`|
|Recording name| 録画名|`Recording4`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:19:39.048`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:19:39.048`|
|Recording duration| 録画時間|`307137`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`3`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.16`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`45`pixels|
|Average validation accuracy | 平均バリデーション精度|`11.6`mm|
|Average validation precision SD | 平均バリデーション予測SD|`2.9`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`3.6`mm|
|Average validation accuracy |平均バリデーション精度| `1.07`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.27`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.33`degrees|
|Average validation accuracy |平均バリデーション精度| `41`pixels|
|Average validation precision SD |平均バリデーション予測SD| `10`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `12`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.003`|
|Presented Media name|提示されたメディア名| `視線誘導問題.003.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [34]:
cleaned_data["視線誘導問題.003"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
12788       12788            229631258          7236684517   
12789       12789            229647867          7236701126   
12790       12790            229664793          7236718052   
12791       12791            229681122          7236734381   
12792       12792            229697776          7236751035   
...           ...                  ...                 ...   
14786       14786            262684551          7269737810   
14788       14788            262700847          7269754106   
14789       14789            262717948          7269771207   
14792       14792            262734469          7269787728   
14793       14793            262750769          7269804028   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
12788          7.254973e+09         872.0         386.0              858.0   
12789          7.254989e+09         872.0         388.0              852.0   
12790          7.255006e+09         865.0         395.0              848.0   
12791          7.255022e+09         866.0         390.0              846.0   
12792          7.255039e+09         867.0         387.0              842.0   
...                     ...           ...           ...                ...   
14786          7.288024e+09           NaN           NaN                NaN   
14788          7.288040e+09           NaN           NaN                NaN   
14789          7.288057e+09         823.0         779.0              876.0   
14792          7.288074e+09         830.0         761.0              818.0   
14793          7.288090e+09         842.0         679.0              839.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
12788              386.0               886.0               386.0  ...   
12789              392.0               891.0               384.0  ...   
12790              398.0               882.0               391.0  ...   
12791              399.0               886.0               382.0  ...   
12792              397.0               891.0               376.0  ...   
...                  ...                 ...                 ...  ...   
14786                NaN                 NaN                 NaN  ...   
14788                NaN                 NaN                 NaN  ...   
14789              675.0               771.0               883.0  ...   
14792              703.0               841.0               820.0  ...   
14793              684.0               845.0               674.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
12788                       0.3574                        0.4612   
12789                       0.3633                        0.4642   
12790                       0.3687                        0.4595   
12791                       0.3693                        0.4617   
12792                       0.3680                        0.4640   
...                            ...                           ...   
14786                          NaN                           NaN   
14788                          NaN                           NaN   
14789                       0.6253                        0.4013   
14792                       0.6510                        0.4381   
14793                       0.6329                        0.4403   

       Gaze point right Y (MCSnorm)  Eye movement type  \
12788                        0.3571           Fixation   
12789                        0.3551           Fixation   
12790                        0.3622           Fixation   
12791                        0.3534           Fixation   
12792                        0.3485           Fixation   
...                             ...                ...   
14786                           NaN       EyesNotFound   
14788                           NaN       EyesNotFound   
14789                        0.8177       Unclassified   
14792                        0.7590       Unclassif

In [35]:
cleaned_data["視線誘導問題.003"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-3/user-3-pic-3.csv")

### カラムの要素が1つしかない変数の説明 写真4
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-3`|
|Recording name| 録画名|`Recording4`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:19:39.048`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:19:39.048`|
|Recording duration| 録画時間|`307137`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`3`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.16`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`45`pixels|
|Average validation accuracy | 平均バリデーション精度|`11.6`mm|
|Average validation precision SD | 平均バリデーション予測SD|`2.9`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`3.6`mm|
|Average validation accuracy |平均バリデーション精度| `1.07`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.27`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.33`degrees|
|Average validation accuracy |平均バリデーション精度| `41`pixels|
|Average validation precision SD |平均バリデーション予測SD| `10`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `12`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.004`|
|Presented Media name|提示されたメディア名| `視線誘導問題.004.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [36]:
cleaned_data["視線誘導問題.004"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
15092       15092            267776847          7274830106   
15093       15093            267793594          7274846853   
15094       15094            267809825          7274863084   
15095       15095            267826765          7274880024   
15096       15096            267843526          7274896785   
...           ...                  ...                 ...   
17129       17129            301827369          7308880628   
17130       17130            301844608          7308897867   
17131       17131            301861199          7308914458   
17133       17133            301877292          7308930551   
17134       17134            301894481          7308947740   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
15092          7.293116e+09         858.0         466.0              860.0   
15093          7.293133e+09         857.0         468.0              848.0   
15094          7.293149e+09         858.0         466.0              850.0   
15095          7.293166e+09         859.0         463.0              853.0   
15096          7.293182e+09         853.0         464.0              840.0   
...                     ...           ...           ...                ...   
17129          7.327164e+09         873.0         691.0              828.0   
17130          7.327181e+09           NaN           NaN                NaN   
17131          7.327198e+09           NaN           NaN                NaN   
17133          7.327214e+09           NaN           NaN                NaN   
17134          7.327231e+09           NaN           NaN                NaN   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
15092              461.0               856.0               471.0  ...   
15093              469.0               866.0               467.0  ...   
15094              467.0               865.0               464.0  ...   
15095              460.0               864.0               466.0  ...   
15096              463.0               867.0               466.0  ...   
...                  ...                 ...                 ...  ...   
17129              664.0               919.0               717.0  ...   
17130                NaN                 NaN                 NaN  ...   
17131                NaN                 NaN                 NaN  ...   
17133                NaN                 NaN                 NaN  ...   
17134                NaN                 NaN                 NaN  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
15092                       0.4271                        0.4459   
15093                       0.4343                        0.4511   
15094                       0.4329                        0.4503   
15095                       0.4260                        0.4500   
15096                       0.4283                        0.4515   
...                            ...                           ...   
17129                       0.6151                        0.4786   
17130                          NaN                           NaN   
17131                          NaN                           NaN   
17133                          NaN                           NaN   
17134                          NaN                           NaN   

       Gaze point right Y (MCSnorm)  Eye movement type  \
15092                        0.4363           Fixation   
15093                        0.4320           Fixation   
15094                        0.4299           Fixation   
15095                        0.4317           Fixation   
15096                        0.4315           Fixation   
...                             ...                ...   
17129                        0.6640            Saccade   
17130                           NaN       EyesNotFound   
17131                           NaN       EyesNotFound   
17133                           NaN       EyesNotFo

In [38]:
cleaned_data["視線誘導問題.004"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-3/user-3-pic-4.csv")